File: JavaScript/JavaScript-Fundamentals/Exam preparation/2012-2013/First task/Sequence.js
function Solve(input) {
	var result = 1;

	// casting from array from string to numbers
	input = input.map(Number);
	// if var i = 2 - that's how we dont take the input N
	for (var i = 2; i < input.length; i++) {
		if (input[i - 1] > input[i]) {
			result++;
		}
	}
	return result;
}

var input = ['9', // this is the input!
			 '1', '8', '8', '7', '6', '5', '7', '7', '6'];
//['9','1', '8', '8', '7', '6', '5', '7', '7', '6']
//['8','7', '1', '2', '-3', '4', '4', '0', '1'];

console.log(Solve(input));

File: Data-Structures-and-Algorithms/02. Linear-Data-Structures/Linear-Data-Structures/Task01/Sequence.cs
namespace Task01
{
    using System;
    using System.Collections.Generic;
    using System.Linq;

    class Sequence
    {
        static void Main()
        {
            // Write a program that reads from the console a sequence of positive integer numbers.
            // The sequence ends when empty line is entered.
            // Calculate and print the sum and average of the elements of the sequence.
            // Keep the sequence in List<int>.

            var numbers = new List<int>();
            Console.WriteLine("Enter a number: ");
            while (true)
            {
                var userInput = Console.ReadLine();
                if (string.IsNullOrEmpty(userInput))
                {
                    break;
                }

                int number = 0;
                var isValidNumber = int.TryParse(userInput, out number);

                if (isValidNumber)
                {
                    numbers.Add(number);
                    Console.WriteLine("Enter a number:");
                }
                else
                {
                    Console.WriteLine("Invalid input!");
                }
            }

            Console.WriteLine("Average of the numbers is: " + numbers.Average());
            Console.WriteLine("The sum of the numbers is: " + numbers.Sum());
        }
    }
}
